debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/transforms/ReadToInsertEvent.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql.transforms;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.data.Envelope;
+import io.debezium.transforms.SmtManager;
+
+public class ReadToInsertEvent<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReadToInsertEvent.class);
+
+    private SmtManager<R> smtManager;
+
+    @Override
+    public R apply(R record) {
+        if (record.value() == null || !smtManager.isValidEnvelope(record)) {
```

Comment:
@gunnarmorling valid Envelope is checked here.

Reply:
👍